core/eolearn/core/eodata.py

```diff
@@ -279,17 +279,25 @@ def timestamp(self, value: list[dt.datetime]) -> None:
         self.timestamps = value
 
     @property
-    def timestamps(self) -> list[dt.datetime]:
+    def timestamps(self) -> list[dt.datetime] | None:
         """A property for handling the `timestamps` attribute."""
         return self._timestamps
 
     @timestamps.setter
-    def timestamps(self, value: Iterable[dt.datetime]) -> None:
-        if isinstance(value, Iterable) and all(isinstance(time, (dt.date, str)) for time in value):
+    def timestamps(self, value: Iterable[dt.datetime] | None) -> None:
+        if value is None:
+            self._timestamps = None
+        elif isinstance(value, Iterable) and all(isinstance(time, (dt.date, str)) for time in value):
```

Comment:
The `isinstance` check here is not in line with the expected type of the setter, it would make more sense if this is `isinstance(time, dt.datetime)`, right? Otherwise the error message should be fixed

Seems like this has worked so far because `isinstance(time, dt.date)` is `True` even if `time` is of the `dt.datetime` type.

Reply:
`dt.date` is technically a superclass of `dt.datetime` (although it is actually not, it's weird)
The error message tells the users the desired input, but here we do some convenience parsing for them (we're all lazy and use `"2021-05-01"` instead of actual datetime objects).

The type of the setter actually doesn't even matter here :see_no_evil: mypy always assumes the type is the same as that in the getter. python....